sdk/src/crypto.rs

```diff
@@ -24,30 +24,103 @@ use crate::{status_code_to_bool, sys, SyscallResult};
 /// NOTE: This only supports f1 and f3 addresses.
 pub fn verify_signature(
     signature: &Signature,
-    signer: &Address,
-    plaintext: &[u8],
+    signers: &[Address],
+    digests: &[&[u8]],
 ) -> SyscallResult<bool> {
+    use fvm_shared::address::{Payload, Protocol};
+
+    let num_signers = signers.len();
+    assert_ne!(
+        num_signers, 0,
+        "no signing addresses were provided for signature verification"
+    );
+    assert_eq!(
+        num_signers,
+        digests.len(),
+        "unequal numbers of signers and digests"
+    );
```

Comment:
Please reserve panics for things that are statically impossible, or well documented edge-cases where returning an error would add too much noise.

In this case, we're likely handling user input so panicing is almost never the right answer.